### PR TITLE
IO-1858: withModel and withNewModel HOC updates, create Model and NewModel

### PR DESCRIPTION
**Pull request by jmadson**
- `withModel` and `withNewModel` HOCs now support setting the `modelName` dynamically as a prop
- `Model` and `NewModel` render props components created

### CODE REVIEW

File: src/Model.test.tsx [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+import * as React from "react";
+
+import {
+  fakeModelModule,
+  initializeTestServices,
+  seedService,
+} from "redux-data-service";
+
+import { spy, stub } from "sinon";
```
Comment (JZiemnicki) [inline src/Model.test.tsx]: Imports in import declaration are not used. (spy and stub).
Remove import from sinon.

File: src/Model.test.tsx [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+import { lorem, random } from "faker";
```
Comment (JZiemnicki) [inline src/Model.test.tsx]: Import in import declaration are not used. (lorem and random)
Remove import from faker.

File: src/NewModel.test.tsx [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+import * as React from "react";
+
+import {
+  fakeModelModule,
+  initializeTestServices,
+  FakeModel,
+} from "redux-data-service";
+
+import { spy, stub } from "sinon";
```
Comment (JZiemnicki) [inline src/NewModel.test.tsx]: Imports from Sinon can be removed. They are not used.

File: src/NewModel.test.tsx [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+import { lorem, random } from "faker";
```
Comment (JZiemnicki) [inline src/NewModel.test.tsx]: Imports from faker are not used and can be removed.

File: src/WithModel.test.tsx [hunk 5]
```diff
@@ -108,30 +129,64 @@ describe("withModel", () => {
       const stubSubscribe = stub(fakeModelObservable, "subscribe")
```
Comment (JZiemnicki) [inline src/WithModel.test.tsx]: Unused constant **stubSubscribe**
Can be removed

File: src/WithModel.test.tsx [hunk 6]
```diff
@@ -1,10 +1,11 @@
-// tslint:disable:max-classes-per-file no-unused-expression
+// tslint:disable: no-unused-expression
 
 import * as React from "react";
 import { fakeModelModule, getDataService, initializeTestServices, seedService } from "redux-data-service";
 
 import { Subject } from "rxjs/Subject";
 import { spy, stub } from "sinon";
+import { random } from "faker";
```
Comment (JZiemnicki) [inline src/WithModel.test.tsx]: Import can be removed. Not using faker.

File: src/WithModel.test.tsx [hunk 7]
```diff
@@ -1,10 +1,11 @@
-// tslint:disable:max-classes-per-file no-unused-expression
+// tslint:disable: no-unused-expression
 
 import * as React from "react";
 import { fakeModelModule, getDataService, initializeTestServices, seedService } from "redux-data-service";
```
Comment (JZiemnicki) [inline src/WithModel.test.tsx]: Maybe consider breaking apart onto separate lines for readability?

File: src/WithModel.ts [hunk 8]
```diff
@@ -1,34 +1,45 @@
 import "rxjs/add/operator/switchMap";
 import "rxjs/add/operator/combineLatest";
 
-import {Observable} from "rxjs/Observable";
+import { Observable } from "rxjs/Observable";
 import rxjsConfig from "recompose/rxjsObservableConfig";
-import {branch, ComponentEnhancer, compose, mapPropsStreamWithConfig} from "recompose";
+import { branch, ComponentEnhancer, compose, defaultProps, mapProps, mapPropsStreamWithConfig } from "recompose";
```
Comment (JZiemnicki) [inline src/WithModel.ts]: Maybe consider breaking apart onto separate lines for readability?

File: yarn.lock [hunk 9]
```diff
@@ -97,7 +97,6 @@
 "@types/anymatch@*":
```
Comment (JZiemnicki) [inline yarn.lock]: I am not sure if you intentionally wanted to change yarn.lock. If not maybe undo changes to it so it is not part of the pull request?

Comment (jmadson) [reply yarn.lock]: This was due to changing `package.json` (I changed the hard version pin on `redux-data-service` from `0.0.16` to `0.0.32`, the most recent release). When we update `package.json`, we do want to include the associated changes in the `yarn.lock` file.

File: src/NewModel.tsx [hunk 10]
```diff
@@ -0,0 +1,11 @@
...
+import { withRenderProps } from "recompose";
+
+import { IWithModelProps } from "./WithModel";
+import { withNewModel } from "./WithNewModel";
+
+export interface INewModelProps extends IWithModelProps {
+  [key: string]: any;
+}
+
+export const NewModel = withRenderProps<INewModelProps>(withNewModel());
```
[the remark this answers is not in the document]
Comment (jonsca) [reply src/NewModel.tsx]: For posterity, we are doing this so that we have something to find within the wrapper in the test.
